prov/efa/src/rxr/rxr_read.c

```diff
@@ -264,8 +264,6 @@ int rxr_read_init_iov(struct rxr_ep *ep,
 
 		for (i = 0; i < tx_entry->iov_count; ++i) {
 			assert(tx_entry->mr[i]);
-			read_iov[i].addr = (uint64_t)tx_entry->iov[i].iov_base;
-			read_iov[i].len = tx_entry->iov[i].iov_len;
```

Comment:
Let's do memory registration for shm regardless shm_cma_cap, eg:

```
if (tx_entry->desc[0]) {
     read_iov[I].key = fi_rma_key(desc)
} else{
    memory registration
    read_iov[I].key = fi_rma_key(mr[I])
}
```
Because this is the correct way to do things, and also get ride of shm_cma_cap altogether.

Reply:
But that would affect performance as we are doing unnecessary mr registration.

Reply:
With MR cache, I do not think it will have big impact. 

But your point is also valid, let us keep it this way, and run some tests to determine if we can do what I suggested.